### PR TITLE
EIP1-2471: EIP1-2782: Map supporting information format to db entity

### PR DESCRIPTION
**Pull request by adacayi**
This PR is to map the supporting information format in the `SendApplicationToPrintMessage` model to db entity to persist that value.

### CODE REVIEW

**Review by Neil-Massey — APPROVED** [review]
LGTM

File: build.gradle.kts [hunk 1]
```diff
@@ -133,6 +133,7 @@ tasks.withType<KotlinCompile> {
 tasks.withType<Test> {
     dependsOn(tasks.withType<GenerateTask>())
     useJUnitPlatform()
+    jvmArgs("--add-opens", "java.base/java.time=ALL-UNNAMED")
```
Comment (Neil-Massey) [inline build.gradle.kts:136]: Why was this necessary?

Comment (adacayi) [reply build.gradle.kts:136]: Sorry, @Neil-Massey, I forgot to add the explanation for this. 
Without this, we get an error when we want to spy `java.time` objects:

image

I needed to spy a `Clock` instance in the `PrintRequestMapperTest` to use `@InjectMocks` instead of using reflection to set the dependencies in the `PrintRequestMapper` hence added the above change.

File: src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintRequestMapperTest.kt [hunk 2]
```diff
@@ -46,51 +49,32 @@ class PrintRequestMapperTest {
...
-                property = "Pencadlys Cyngor Gwynedd",
-                street = "Stryd y Jêl",
-                town = "Caernarfon",
-                area = "Gwynedd",
-                postcode = "LL55 1SH",
-            )
-        )
+        given(supportingInformationFormatMapper.toPrintRequestEntityEnum(any()))
+            .willReturn(supportingInformationFormatEntityEnum)
+        val expectedEnglishEroContactDetails = toEroContactDetails(ero.englishContactDetails)
```
Comment (adacayi) [inline src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintRequestMapperTest.kt:76]: I changed these, so it is more explicit that the expected values are derived from the `eroDto` and is easier to read and understand.

File: build.gradle.kts (hunk 1 above)
Comment (Neil-Massey) [reply build.gradle.kts:136]: Thanks @abdullah-sanver-valtech - I wondered if that may have bee the reason but wasn't sure 👍

File: src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintRequestMapperTest.kt (hunk 2 above)
Comment (adacayi) [reply src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintRequestMapperTest.kt:76]: Thanks, @Neil-Massey , for your suggestion.
Added it as an extension method for tests.

<pre>
val expectedEnglishEroContactDetails = ero.englishContactDetails.toElectoralRegistrationOffice()
val expectedWelshEroContactDetails = ero.welshContactDetails!!.toElectoralRegistrationOffice()
</pre>

Comment (adacayi) [reply src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintRequestMapperTest.kt:76]: <pre>
package uk.gov.dluhc.printapi.testsupport.testdata.dto

fun EroContactDetailsDto.toElectoralRegistrationOffice() =
    ElectoralRegistrationOffice(
        name = name,
        phoneNumber = phoneNumber,
        website = website,
        emailAddress = emailAddress,
        address = with(address) {
            Address(
                property = property,
                street = street,
                town = town,
                area = area,
                postcode = postcode,
            )
        }
    )
</pre>
